robots/purePursuitController.h

```diff
@@ -0,0 +1,169 @@
+/*
+    Pure Pursuit Controller. The controller takes a list of waypoints as a path to follow,
+    then it calculates the optimal turning angle to stay on the path
+*/
+
+// BoB robotics includes
+#include "common/main.h"
+#include "robots/car_display.h"
+
+// Third-party includes
+#include "third_party/units.h"
+
+// Standard C++ includes
+#include <chrono>
+#include <iostream>
+#include <thread>
+
+// sign function
+template <typename T> int sgn(T val) {
+    return (T(0) < val) - (val < T(0));
+}
+
+namespace BoBRobotics {
+namespace Robots {
+
+using namespace units::math;
```

Comment:
using namespaces in headers is generally a bad idea as they 'pollute' anything that includes the header. For instance, if I include this and am writing my own code in ``BoBRobotics::Robots``, I wouldn't expect ``sin`` to do something different than normal. In this case, you are only using the literals in ``getTurningAngle`` and the maths functions in ``getLookAheadPoint`` and ``computeTurningAngle`` so you only need to put those ``using`` statements there.

Reply:
Agreed. But just to add that I think the ``using namespace units::literals;`` below is fine. String literals are a bit of a special case, I feel.